### PR TITLE
Add missing patterns to .gitignore

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
A few files creeped into the repo since https://github.com/digital-asset/daml/pull/6324

```
> git status
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	.ijwb/
	compatibility/.bazel-cache/
	compatibility/.ijwb/
	daml-ledger-0.0.0.tgz
	daml-react-0.0.0.tgz
	daml-types-0.0.0.tgz
	navigator/backend/scenarios/rental/.interfaces/
	navigator/backend/scenarios/rental/dist/
	target/
```

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,7 +49,7 @@ node_modules/
 
 # IntelliJ IDEA
 .idea
-.ijwb # Bazel
+.ijwb
```
Comment (SamirTalwar-DA) [inline .gitignore:52]: I wonder if this is down to Git version? This works fine for me. (Ship it though.)

File: .gitignore [hunk 2]
```diff
@@ -62,3 +62,12 @@ out
...
 # Visual Studio Code
 .vscode
+
+### Artifacts ###
+
+# Artifacts
+*.tgz
+target
+dist
+.interfaces
```
Comment (SamirTalwar-DA) [inline .gitignore:72]: I think I'd like to see this near the top. Perhaps rename "Log files" to "Artifacts" and move them there?

(And sort them please.)

File: .gitignore (hunk 1 above)
Comment (stefanobaghino-da) [reply .gitignore:52]: Could be.

File: .gitignore (hunk 2 above)
Comment (stefanobaghino-da) [reply .gitignore]: Addressed by https://github.com/digital-asset/daml/pull/6348/commits/ab00a4dcab773ba71746934783f82b1b88669b02

File: .gitignore (hunk 1 above)
Comment (garyverhaegen-da) [reply .gitignore:56]: Official docs only mention lines starting with # as comments.

**Pull request merged**